hazelcast-code-generator/src/main/java/com/hazelcast/client/protocol/generator/CodeGenerationUtils.java

```diff
@@ -47,6 +46,30 @@
         put("com.hazelcast.map.impl.SimpleEntryView", "Hazelcast.Map.SimpleEntryView");
     }};
 
+    private static final Map<String, String> JAVA_TO_CPP_TYPES = new HashMap<String, String>() {{
+        put("java.lang.Integer", "int32_t");
+        put("int", "int32_t");
+        put("boolean", "bool");
+        put("java.lang.Boolean", "bool");
+        put("short", "int16_t");
+        put("char", "int8_t");
+        put("byte", "uint8_t");
+        put("long", "int64_t");
+        put(DATA_FULL_NAME, "serialization::pimpl::Data");
+        put("java.lang.String", "std::string");
+        put("byte[]", "std::vector<byte>");
+        put("java.util.List", "std::vector");
+        put("java.util.Set", "std::vector");
+        put("java.util.Collection", "std::vector");
+        put("java.util.Map", "std::vector<std::pair");
```

Comment:
No, it was intentional, the closing bracket is inserted at the end after the inner types in getLanguageType method. It looks weird but it was the easiest way for me.